### PR TITLE
Added Mac differences for build dependencies and environment Ref #14

### PR DESCRIPTION
**Pull request by emmonspired**

### CODE REVIEW

File: docs/setting-up-your-environment.md
Comment (harding) [inline docs/setting-up-your-environment.md]: I think it's unclear where this MacOS section ends, e.g.:



I suggest combining this information down a bit to under the original "Server running..." line, possibly combining it with the final Mac Differences section in your diff.

File: docs/setting-up-your-environment.md
Comment (harding) [inline docs/setting-up-your-environment.md]: Can you still access your server on MacOS at http://0.0.0.0:4000/ ?  (I think that should work even outside Linux, but I don't know for sure.)  If so, this difference might not need to be mentioned, but I'm fine with keeping it in if you think it's useful.

**Review by harding — CHANGES_REQUESTED** [review]
Thanks for figuring this all out!  One requested change and one nitpick.

File: docs/setting-up-your-environment.md
Comment (kek-coin) [inline docs/setting-up-your-environment.md]: On my (non-Debian) linux system, trying to build also gives the error

```
  Conversion error: Jekyll::Converters::Scss encountered an error while converting 'css/main.scss':
                    Invalid US-ASCII character "\xE2" on line 1917
jekyll 3.0.1 | Error:  Invalid US-ASCII character "\xE2" on line 1917
```

So it seems this is not OSX specific, and therefore should be moved outside of the "Mac differences" part.

Comment (kek-coin) [reply docs/setting-up-your-environment.md]: Also I would suggest changing the instructions to run
```bash
export LC_ALL="en_US.UTF-8"
export LANG="en_US.UTF-8"
```
before the `make` command. No need to put that in `.bashrc` or `.zshrc` for a project-specific fix.

Comment (harding) [reply docs/setting-up-your-environment.md]: I think that's a bug that should be dealt with separately (I'll look into it).  The makefile already should already be running locale independently:

```make
## Always build using the default locale so log messages can be grepped.
## This should not affect webpage output.
build:
        $S export LANG=C.UTF-8 ...
```

File: docs/setting-up-your-environment.md
Comment (emmonspired) [reply docs/setting-up-your-environment.md]: I just tested it - yes, that 0.0.0.0:4000 address is accessible, but it wasn't apparent from the console output. I think it's worth keeping in.

File: docs/setting-up-your-environment.md
Comment (emmonspired) [reply docs/setting-up-your-environment.md]: Thought it was only my local environment, worth mentioning. Should I create a new issue for this?

File: docs/setting-up-your-environment.md
Comment (emmonspired) [reply docs/setting-up-your-environment.md]: Ok, thanks for the feedback. Let me give it a second try to consolidate it.

Comment (emmonspired) [reply docs/setting-up-your-environment.md]: Actually, as per @kek-coin 's comments below I'll create a new issue for this section.

File: docs/setting-up-your-environment.md
Comment (harding) [reply docs/setting-up-your-environment.md]: I think I have it fixed here; if you can test, that'd be great.  https://github.com/achow101/btcinformation.org/pull/19

Comment (emmonspired) [reply docs/setting-up-your-environment.md]: I removed the bug section from my original pull request. Please review.

**Review by emmonspired — COMMENTED** [review]
I believe I removed the section regarding the en-US bug.

**Comment by emmonspired** [comment]
Please have a new look @harding @kek-coin

File: docs/setting-up-your-environment.md
Comment (kek-coin) [inline docs/setting-up-your-environment.md]: Nit: cleanup whitespace.

File: docs/setting-up-your-environment.md
Comment (kek-coin) [inline docs/setting-up-your-environment.md]: Grammar: "in replace of" should be "in place of".

File: docs/setting-up-your-environment.md
Comment (kek-coin) [inline docs/setting-up-your-environment.md]: It also looks like this to me on linux, so not OSX-specific; consider removing. "A message *like* this" (emphasis mine) should already cover any different server ip or port anyway.

**Review by kek-coin — CHANGES_REQUESTED** [review]

File: docs/setting-up-your-environment.md
Comment (achow101) [inline docs/setting-up-your-environment.md]: No need to have a header here to mark the end of this section.

File: docs/setting-up-your-environment.md
Comment (achow101) [inline docs/setting-up-your-environment.md]: You can just have this as "Mac OSX Systems" and add a header above titled "Ubuntu/Debian Systems"

File: docs/setting-up-your-environment.md
Comment (achow101) [reply docs/setting-up-your-environment.md]: I agree that this is unnecessary.

**Review by achow101 — COMMENTED** [review]
Please squash your commits.

File: docs/setting-up-your-environment.md
Comment (emmonspired) [reply docs/setting-up-your-environment.md]: I'll make the change.

File: docs/setting-up-your-environment.md
Comment (emmonspired) [reply docs/setting-up-your-environment.md]: will remove and take your suggestions into account for headers, as well as squash the commits

File: docs/setting-up-your-environment.md
Comment (emmonspired) [reply docs/setting-up-your-environment.md]: No problem.

File: docs/setting-up-your-environment.md
Comment (emmonspired) [reply docs/setting-up-your-environment.md]: ok

File: docs/setting-up-your-environment.md
Comment (harding) [inline docs/setting-up-your-environment.md]: This needs to go before the preceding paragraph.

File: docs/setting-up-your-environment.md
Comment (harding) [inline docs/setting-up-your-environment.md]: These extra spaces should be deleted.

**Review by harding — CHANGES_REQUESTED** [review]

File: docs/setting-up-your-environment.md
Comment (emmonspired) [reply docs/setting-up-your-environment.md]: Well, gives me a chance to figure out squashing again :) I can't do it graphically on the site so i am doing it command line.

**Comment by emmonspired** [comment]
Please review 2575539 so we can close this once and for all 👍

**Review by emmonspired — COMMENTED** [review]
I believe everything looks good on my end. Please review @harding

**Comment by achow101** [comment]
ACK 25755392d0e38eea94db1819e6c509f4ca0e2d4a

**Comment by emmonspired** [comment]
Ugh, I made a commit that is waiting to be merged, but can’t because I didn’t have the GPG signature in place. Any way to sign it after the fact @achow101 ?

**Comment by achow101** [comment]
I think I can actually still merge it. I was going to do an experiment with this PR to see whether I can. If not, I will actually disable the require signed commits protection because I think it is unreasonable to expect everyone to sign their commits. The build server does check for GPG signatures on the head commit. I was actually waiting for someone else to ACK this before I merge it.

**Comment by emmonspired** [comment]
I actually got the GPG going locally just now and I was going to re-base the entire thing again, but I *really* don't want to go through the hassle for this minor of a change @achow101

**Comment by emmonspired** [comment]
LMK what you want me to do. I can leave it as is or resubmit. @achow101

**Comment by emmonspired** [comment]
@achow101 I hope you don't mind but I was talking with @kek-coin in the Bitcoin core #tech-chat and we both came to the conclusion that I could use more practice with rebase, so I did, and now it's 45a2baa verified. Also gave me a chance to check that GPG worked.

**Comment by kek-coin** [comment]
ACK 45a2baa.

As an aside, the discussion we had in slack was about various `git` workflows, didn't intend for it to meddle with @achow101's experiment. Perhaps that can be done in some next PR.

**Comment by emmonspired** [comment]
I have another addition to make since I only checked for `make preview` to work. Turns out the `make all` had failed because Mac OSX is missing -d flag for Xargs.

As per... 
https://superuser.com/questions/467176/replacement-for-xargs-d-in-osx

brew install findutils
PATH="/usr/local/opt/findutils/libexec/gnubin:$PATH"

**Comment by emmonspired** [comment]
Also - if you're wondering why i put that PATH reference in there, it's a hint that comes from the output of the `brew install findutils` @kek-coin 

I guess alternatively there was the "with-default-names" option but I wanted to choose the safer route, imo.

```
demmonss-MacBook-Pro:btc-test demmons$ brew reinstall findutils
==> Reinstalling findutils 
==> Downloading https://homebrew.bintray.com/bottles/findutils-4.6.0.high_sierra.bottle.2.tar.gz
Already downloaded: /Users/demmons/Library/Caches/Homebrew/findutils--4.6.0.high_sierra.bottle.2.tar.gz
==> Pouring findutils--4.6.0.high_sierra.bottle.2.tar.gz
==> Caveats
All commands have been installed with the prefix 'g'.
If you do not want the prefix, install using the "with-default-names" option.

If you need to use these commands with their normal names, you
can add a "gnubin" directory to your PATH from your bashrc like:

    PATH="/usr/local/opt/findutils/libexec/gnubin:$PATH"

Additionally, you can access their man pages with normal names if you add
the "gnuman" directory to your MANPATH from your bashrc as well:

    MANPATH="/usr/local/opt/findutils/libexec/gnuman:$MANPATH"
==> Summary
🍺  /usr/local/Cellar/findutils/4.6.0: 34 files, 1.6MB
```

**Comment by achow101** [comment]
ACK d2b6339029e6c1e54de32bd31728330d8511c57c

**Comment by kek-coin** [comment]
ACK d2b6339. Was a bit hesitant about the PATH thing - personally I am not a big fan of having to set/override environment variables to build a project since it's prone to error, but in this case it seems like the sanest/safest option.

**Comment by emmonspired** [comment]
Agreed @kek-coin ... I debated the pros/cons and wanted to make sure I wasn't going to suggest anyone overwrite any local binaries, and a simple path fix is safer.

**Pull request merged**
